sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DocumentCollection.java

```diff
@@ -427,12 +430,18 @@ public DocumentCollection getWrappedItem() {
 
         private void writeObject(ObjectOutputStream objectOutputStream) throws IOException {
             documentCollection.populatePropertyBag();
-            objectOutputStream.writeObject(documentCollection.getPropertyBag());
+            ObjectNode docCollectionNode = OBJECT_MAPPER.createObjectNode();
+            docCollectionNode.set("col", documentCollection.getPropertyBag());
+            docCollectionNode.set("altLink", TextNode.valueOf(documentCollection.getAltLink()));
+            objectOutputStream.writeObject(docCollectionNode);
         }
 
         private void readObject(ObjectInputStream objectInputStream) throws IOException, ClassNotFoundException {
             ObjectNode objectNode = (ObjectNode) objectInputStream.readObject();
-            this.documentCollection = new DocumentCollection(objectNode);
+            ObjectNode collectionNode = (ObjectNode)objectNode.get("col");
```

Comment:
How does it work if it's deserializing an older version?

Reply:
It isn't - and doesn't have to - because in a Spark cluster deployment happens as offline deployment